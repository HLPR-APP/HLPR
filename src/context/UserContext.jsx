import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
