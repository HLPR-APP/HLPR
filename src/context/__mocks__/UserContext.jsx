import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
