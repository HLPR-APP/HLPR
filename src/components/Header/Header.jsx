import {
  Box,
  Flex,
  Avatar,
  Button,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
  MenuDivider,
  useDisclosure,
  useColorModeValue,
  Stack,
  useColorMode,
  Center,
} from '@chakra-ui/react';
import { MoonIcon, SunIcon } from '@chakra-ui/icons';
import { useUser } from '../../context/UserContext.jsx';
import { Link } from 'react-router-dom';
import { signOutUser } from '../../services/users.js';
import { useHistory } from 'react-router-dom';

// const NavLink = ({ children }) => (
//   <Link
//     px={2}
//     py={1}
//     rounded={'md'}
//     _hover={{
//       textDecoration: 'none',
//       bg: useColorModeValue('gray.200', 'gray.700'),
//     }}
//     href={'#'}
//   >
//     {children}
//   </Link>
// );

export default function Header() {
  const { colorMode, toggleColorMode } = useColorMode();
  const { isOpen, onOpen, onClose } = useDisclosure();

  // export default function HeaderLogin({ loggedin }) {
  const { user, setUser } = useUser();
  const history = useHistory();

  const handleLogOut = async () => {
    await signOutUser();
    setUser(null);
    history.push('/');
  };

  const loggedin = user ? true : false;

  const loggedInDiv = (
    <Box>
      {user?.email} <Button onClick={handleLogOut}>sign out</Button>
    </Box>
  );

  const notLoggedInDiv = (
    <Box>
      {/* <p>Not Signed in</p> */}
      <Link to="/">Sign In</Link>
    </Box>
  );

  return (
    <>
      <Box bg={useColorModeValue('gray.100', 'gray.900')} px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <Box>
            <Button>
              <Link to="/tasklist">HLPR</Link>
            </Button>
          </Box>

          <Flex alignItems={'center'}>
            <Stack direction={'row'} spacing={7}>
              <Button onClick={toggleColorMode}>
                {colorMode === 'light' ? <MoonIcon /> : <SunIcon />}
              </Button>

              <Menu>
                <MenuButton
                  as={Button}
                  rounded={'full'}
                  variant={'link'}
                  cursor={'pointer'}
                  minW={0}
                >
                  <Avatar
                    size={'sm'}
                    src={'https://avatars.dicebear.com/api/male/username.svg'}
                  />
                </MenuButton>
                <MenuList alignItems={'center'}>
                  <br />
                  <Center>
                    <Avatar
                      size={'2xl'}
                      src={'https://avatars.dicebear.com/api/male/username.svg'}
                    />
                  </Center>
                  <br />
                  <Center>{user?.email && <p>{user.email}</p>}</Center>
                  <br />
                  <MenuDivider />
                  {user?.email && (
                    <MenuItem onClick={handleLogOut}>Sign Out</MenuItem>
                  )}
                  {!loggedin && notLoggedInDiv}
                  {/* <MenuItem>{loggedin ? loggedInDiv : notLoggedInDiv}</MenuItem> */}
                  {/* <MenuItem>Your Servers</MenuItem> */}
                  {user?.email && (
                    <MenuItem>
                      <Link to="/profile">Profile</Link>
                    </MenuItem>
                  )}
                  <MenuDivider />
                  <MenuItem>
                    <Link to="/aboutus">About Us</Link>
                  </MenuItem>
                  {/* <MenuItem>Logout</MenuItem> */}
                </MenuList>
              </Menu>
            </Stack>
          </Flex>
        </Flex>
      </Box>
    </>
  );
}
