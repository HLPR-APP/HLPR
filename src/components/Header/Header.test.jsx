import dotenv from 'dotenv';
dotenv.config();
import React from 'react';
import { render } from '@testing-library/react';
import { UserProvider } from '../../context/UserContext.jsx';
import Header from './Header.jsx';
import { MemoryRouter } from 'react-router-dom';

it('should render header', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
