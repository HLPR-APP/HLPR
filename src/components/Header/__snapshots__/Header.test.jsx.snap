// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header 1`] = `
<div>
  <div
    class="css-1m65r4j"
  >
    <div
      class="css-jfo7z"
    >
      <div
        class="css-0"
      >
        <button
          class="chakra-button css-5c2176"
          type="button"
        >
          <a
            href="/tasklist"
          >
            HLPR
          </a>
        </button>
      </div>
      <div
        class="css-70qvj9"
      >
        <div
          class="chakra-stack css-r9wgs2"
        >
          <button
            class="chakra-button css-5c2176"
            type="button"
          >
            <svg
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="5"
                />
                <path
                  d="M12 1v2"
                />
                <path
                  d="M12 21v2"
                />
                <path
                  d="M4.22 4.22l1.42 1.42"
                />
                <path
                  d="M18.36 18.36l1.42 1.42"
                />
                <path
                  d="M1 12h2"
                />
                <path
                  d="M21 12h2"
                />
                <path
                  d="M4.22 19.78l1.42-1.42"
                />
                <path
                  d="M18.36 5.64l1.42-1.42"
                />
              </g>
            </svg>
          </button>
          <button
            aria-controls="menu-list-4"
            aria-expanded="false"
            aria-haspopup="menu"
            class="chakra-button chakra-menu__menu-button css-1t30i5w"
            id="menu-button-4"
            type="button"
          >
            <span
              class="css-xl71ch"
            >
              <span
                class="chakra-avatar css-1ruhx4w"
              >
                <svg
                  aria-label=" avatar"
                  class="chakra-avatar__svg css-16ite8i"
                  role="img"
                  viewBox="0 0 128 128"
                >
                  <path
                    d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                    fill="currentColor"
                  />
                  <path
                    d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
          <div
            class="css-0"
            style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
          >
            <div
              aria-orientation="vertical"
              class="chakra-menu__menu-list css-1jabnse"
              id="menu-list-4"
              role="menu"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
              tabindex="-1"
            >
              <br />
              <div
                class="css-gmuwbf"
              >
                <span
                  class="chakra-avatar css-1ruhx4w"
                >
                  <svg
                    aria-label=" avatar"
                    class="chakra-avatar__svg css-16ite8i"
                    role="img"
                    viewBox="0 0 128 128"
                  >
                    <path
                      d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                      fill="currentColor"
                    />
                    <path
                      d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <br />
              <div
                class="css-gmuwbf"
              />
              <br />
              <hr
                aria-orientation="horizontal"
                class="chakra-menu__divider css-0"
                role="separator"
              />
              <div
                class="css-9n3bbc"
              >
                <a
                  href="/"
                >
                  Sign In
                </a>
              </div>
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="0"
                id="menu-list-4-menuitem-1"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <a
                  href="/tasklist"
                >
                  Task List
                </a>
              </button>
              <hr
                aria-orientation="horizontal"
                class="chakra-menu__divider css-0"
                role="separator"
              />
              <button
                class="chakra-menu__menuitem css-1k756t9"
                data-index="1"
                id="menu-list-4-menuitem-2"
                role="menuitem"
                tabindex="-1"
                type="button"
              >
                <a
                  href="/aboutus"
                >
                  About Us
                </a>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
